### ISSUE TITLE
Non-human exclusion

### ISSUE THREAD

**Issue by albullington** [inaturalist/react-native-inat-camera#65]
**Is your feature request related to a problem? Please describe.**
The AR camera sometimes identifies humans as non-humans, which often isn't a positive experience for the end user.

**Describe the solution you'd like**
Tackling this the same way we tackled it [on the web](https://github.com/inaturalist/iNaturalistAPI/pull/242):

For offline results:
- [ ] Exclude non-human predictions when human is the top prediction and is 0.2 threshold points higher than the second result
- [ ] Show an unknown prediction if human is among the predictions but there is no common ancestor (i.e. scores are spread out instead of strongly favoring a higher-level taxon)

**Describe alternatives you've considered**
None

**Additional context**
Basically mimicking [this commit](https://github.com/inaturalist/iNaturalistAPI/pull/242/commits/c4ba229ba551eb2a0247f7070b2fcc918ef5c687)

**Comment by kueda** [comment]
For the record, I don't think react-native-inat-camera can mimic what we're doing in the API since the approach to classifying an image is so different (rnic returns a "best" branch by starting at the root of the tree and always choosing the child with the highest score, while inatapi returns top results from anywhere in the tree as well as a "common ancestor" taxon that contains the highest-scoring results).

So I guess at this point this is more a request for proposals for how rnic might try to achieve the same goal of returning only human or nothing when the photo might be of a human. When human is the top-scoring result I think it will currently work fine: we'll just return a branch that contains human (maybe the client will want to exclude higher ranks in that case). However, when human is a close second or close third or something, we should probably return nothing. Some ideas for handling that:

1. **If human is among the top X results and its score is Y less than the next-best result, show no results.** FWIW, we tried this in the API and it wasn't great, b/c pics where a non-human was the subject but a human was also in the photo often resulted in situations like this, so we'd end up showing no suggestions for common situations like a person holding a fish they just caught or a hand holding a flower.
2. **Calculate the scores for the branch ending in human and compare that with the “best” branch, and if the leaf of the best branch has a score within a X of the equivalent rank on the human branch, return nothing.** Has similar problems to the first idea, though maybe a bit different.
3. **Return the human branch as well as the best branch and let the client decide what to show.** Kind of a punt, but maybe provides more flexibility to the client.
